libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/summary/list/DirectionListAdapter.java

```diff
@@ -69,7 +69,9 @@ public void updateSteps(RouteProgress routeProgress) {
   }
 
   public void clear() {
-    legSteps.remove(legSteps.size());
+    for (int i = 0; i < legSteps.size(); i++) {
+      legSteps.remove(i);
```

Comment:
this for loop isn't needed, you can instead use, legSteps.clear(); to remove all items. Also a foreach is always better to use here. Effective Java: http://thefinestartist.com/effective-java/46